clippy_lints/src/missing_inline.rs

```diff
@@ -173,3 +177,8 @@ impl<'tcx> LateLintPass<'tcx> for MissingInline {
         check_missing_inline_attrs(cx, attrs, impl_item.span, desc);
     }
 }
+
+fn fn_contains_extern(cx: &LateContext<'_>, def_id: DefId) -> bool {
```

Comment:
Small nit: can you add a short comment mentioning what this is for? Just something like "Checks if this function is externally exported, where #[inline] wouldn't have the desired effect and a rustc warning would be triggered, see #15301" should suffice

And maybe rename the function too while we're at it, I assume this function name "fn_contains_extern" is still from before when it was retokenizing the span, but this isn't happening anymore. Maybe "fn_is_externally_exported". What do you think?